feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,420 @@
+		ateConf   *ondatra.ATETopology
+		wantState *telemetry.NetworkInstance_Protocol_Bgp
+	}{
+		{
+			name:    "basic internal",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: dutAS}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: dutAS, peerIP: dutIP, connType: "INTERNAL"}),
+		},
+		{
+			name:    "basic external",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL"}),
+		},
+		{
+			name:    "explicit AS",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, nbrLocalAS: 100}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL"}),
+		},
+		{
+			name:    "explicit router id",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, routerID: dutAttrs.IPv4}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL"}),
+		},
+		{
+			name:    "password",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, authPwd: "AUTHPASSWORD"}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL", authPwd: "AUTHPASSWORD"}),
+		},
+		{
+			name:    "hold-time, keepalive timer",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, holdTime: 100}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL", holdTime: 100}),
+		},
+		{
+			name:    "connect-retry",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, connRetTime: 100}),
+			ateConf: configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL"}),
+		},
+		{
+			name:      "hold time negotiated",
+			dutConf:   bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, holdTime: 100}),
+			ateConf:   configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutIP, connType: "EXTERNAL", holdTime: 135}),
+			wantState: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS, holdTime: 100, negHoldTime: 100}),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+			t.Log("Clear BGP Configs on DUT")
+			dutConfPath.Replace(t, nil)
+			t.Log("Configure BGP Configs on DUT")
+			dutConfPath.Replace(t, tc.dutConf)
+			t.Log("Configure BGP on ATE")
+			tc.ateConf.Push(t)
+			tc.ateConf.StartProtocols(t)
+			t.Log("Verify BGP session state : ESTABLISHED")
+			nbrPath.SessionState().Await(t, time.Second*150, telemetry.Bgp_Neighbor_SessionState_ESTABLISHED)
+			stateDut := statePath.Get(t)
+			wantState := tc.wantState
+			if wantState == nil {
+				// if you don't specify a wantState, we just validate that the state corresponding to each config value is what it should be.
+				wantState = tc.dutConf
+			}
+			confirm.State(t, wantState, stateDut)
```

Comment:
Just this?

```
confirm.State(t, tc.dutConf, statePath.Get(t))
```

Reply:
If wantState == nil, we are assigning it to tc.dutConf.  Otherwise it checks for wantState defined in testcase.
please check below and let me know if this one is good.

	wantState := tc.wantState
			if wantState == nil {
				// if you don't specify a wantState, we just validate that the state corresponding to each config value is what it should be.
				wantState = tc.dutConf
			}
			confirm.State(t, wantState, stateDut)

Reply:
updated these review comments, please check

Reply:
What I don't understand is that wantState looks largely the same as dutConf except it's missing some fields. But is that really necessary? Why can't we just use dutConf to confirm the state?